src/content/en/2020/http2.md

```diff
@@ -4,20 +4,534 @@ part_number: IV
+| 100           | 316.3 | 1237.6 | 83.3 | 2770.7 | 603.0  | 3444.3 |
+
+<figcaption>{{ figure_link(caption="Distribtion of pushed asset sizes (KB) by content type on mobile.", sheets_gid="238923402", sql_file="number_of_h2_and_h3_pushed_resources_and_bytes_by_content_type.sql") }}</figcaption>
+</figure>
+
+Given the low adoption and after measuring how few of the pushed resources are actually useful (match a request that is not already cached), Google has announced the [intent to remove push support from Chrome]( https://groups.google.com/a/chromium.org/g/blink-dev/c/K3rYLvmQUBY/m/vOWBKZGoAQAJ) for both HTTP/2 and gQUIC. Chrome have also not implemented push for HTTP/3.
+
+Despite all these problems, there are circumstances where server push can provide an improvement. The ideal use case is to be able to send a push promise much earlier than the HTML response itself. A scenario where this can benefit is [when a CDN is in use](https://medium.com/@ananner/http-2-server-push-performance-a-further-akamai-case-study-7a17573a3317). The 'dead-time' between the CDN receiving the request and receiving a response from the origin can be used intelligently to warm up the TCP connection and push assets already cached at the CDN.
+
+There was however no standardized method of how to signal to a CDN edge server that an asset should be pushed. Implementations instead reused the preload HTTP link header to indicate this.  This simple approach appears elegant but does not utilise the ‘dead time’ before the HTML is generated unless the headers are sent before the actual content is ready. It triggers the edge to push resources as the HTML is received at the edge which will contend with the  delivery of the HTML.
+
+An alternative proposal being tested is [RFC 8297](https://tools.ietf.org/html/rfc8297) which defines an informative HTTP 103 Early Hints response.  This permits headers to be sent immediately, without having to wait for the server to generate the full response headers.  This can be used by an origin to suggest pushed resources to a CDN, or by a CDN to alert the client to resources that need to be fetched. However, at present, support of this from both a client and server perspective is very low, [but growing](https://www.fastly.com/blog/beyond-server-push-experimenting-with-the-103-early-hints-status-code).
+
+## Getting to a better protocol
+
+Let’s say a client and server support both HTTP/1.1 and HTTP/2. How do they choose which one to use? The most common method is the TLS Application Layer Protocol Negotiation (ALPN). Using this, clients send a list of protocols they support to the server, which picks the one it prefers to use for that connection.  Because the browser needs to negotiate the TLS parameters as part of setting up an HTTPS connection, it can also negotiate the HTTP version at the same time. Since both HTTP/2 and HTTP/1.1 can be served from the same TCP port (443), browsers don’t need to make this selection before opening a connection.
```

Comment:
```suggestion
Let’s say a client and server support both HTTP/1.1 and HTTP/2. How do they choose which one to use? The most common method is TLS Application Layer Protocol Negotiation (ALPN), where clients send a list of protocols they support to the server and it picks the one it prefers to use for that connection.  Because the browser needs to negotiate the TLS parameters as part of setting up an HTTPS connection, it can also negotiate the HTTP version at the same time. Since both HTTP/2 and HTTP/1.1 can be served from the same TCP port (443), browsers don’t need to make this selection before opening a connection.
```